### PR TITLE
[Answer Query Using Materialized Views] Support ORDER BY in origin query

### PR DESCRIPTION
**Pull request by avamingli**
ORDER BY clause could refer result column, result number, table column or expression has table columns.
All that could be processed by aqumv_process_targetlist()
if ORDER BY clause could be computed from materialized views.
```sql
begin;
create table aqumv_t5(c1 int, c2 int, c3 int) distributed by (c1);
insert into aqumv_t5 select i, i+1, i+2 from generate_series(1, 100) i;
insert into aqumv_t5 values (91, NULL, 95);
analyze aqumv_t5;
create incremental materialized view aqumv_mvt5_0 as
  select c1 as mc1, c2 as mc2, c3 as mc3
  from aqumv_t5 where c1 > 90;

set local enable_answer_query_using_materialized_views = on;
explain(costs off, verbose)
select c1, c3 from aqumv_t5 where c1 > 90 order by c2 - c1 - 1 asc;
                                    QUERY PLAN                                    
----------------------------------------------------------------------------------
 Gather Motion 3:1  (slice1; segments: 3)
   Output: mc1, mc3, (((mc2 - mc1) - 1))
   Merge Key: (((mc2 - mc1) - 1))
   ->  Sort
         Output: mc1, mc3, (((mc2 - mc1) - 1))
         Sort Key: (((aqumv_mvt5_0.mc2 - aqumv_mvt5_0.mc1) - 1))
         ->  Seq Scan on public.aqumv_mvt5_0
               Output: mc1, mc3, ((mc2 - mc1) - 1)
 Settings: enable_answer_query_using_materialized_views = 'on', optimizer = 'off'
 Optimizer: Postgres query optimizer
(10 rows)
```
See more examples in test cases.

Authored-by: Zhang Mingli avamingli@gmail.com

fix #ISSUE_Number

---

### Change logs

_Describe your change clearly, including what problem is being solved or what feature is being added._

_If it has some breaking backward or forward compatibility, please clary._

### Why are the changes needed?

_Describe why the changes are necessary._

### Does this PR introduce any user-facing change?

_If yes, please clarify the previous behavior and the change this PR proposes._

### How was this patch tested?

_Please detail how the changes were tested, including manual tests and any relevant unit or integration tests._

### Contributor's Checklist

Here are some reminders and checklists before/when submitting your pull request, please check them:

- [ ] Make sure your Pull Request has a clear title and commit message. You can take [git-commit](https://github.com/cloudberrydb/cloudberrydb/blob/main/.gitmessage) template as a reference.
- [ ] Sign the Contributor License Agreement as prompted for your first-time contribution(*One-time setup*).
- [ ] Learn the [coding contribution guide](https://cloudberrydb.org/contribute/code), including our code conventions, workflow and more.
- [ ] List your communication in the [GitHub Issues](https://github.com/cloudberrydb/cloudberrydb/issues) or [Discussions](https://github.com/orgs/cloudberrydb/discussions) (if has or needed).
- [ ] Document changes.
- [ ] Add tests for the change
- [ ] Pass `make installcheck`
- [ ] Pass `make -C src/test installcheck-cbdb-parallel`
- [ ] Feel free to request `cloudberrydb/dev` team for review and approval when your PR is ready🥳

### CODE REVIEW

File: src/backend/optimizer/plan/aqumv.c [hunk 1]
```diff
@@ -458,11 +458,11 @@ answer_query_using_materialized_views(PlannerInfo *root,
 			/*
 			 * Update pathkeys which may be changed by qp_callback.
 			 * Set belows if corresponding feature is supported.
-			 * sort_pathkeys
 			 * distinct_pathkey
 			 * window_pathkeys
 			 */
 			root->group_pathkeys = subroot->group_pathkeys;
+			root->sort_pathkeys = subroot->sort_pathkeys;
```
Comment (yjhjstz) [inline src/backend/optimizer/plan/aqumv.c:465]: what's diffrence with `query_pathkeys` ?

Comment (avamingli) [reply src/backend/optimizer/plan/aqumv.c:465]: > what's diffrence with `query_pathkeys` ?

`query_pathkeys` is the desired pathkeys for query_planner(),  see details in 
https://github.com/postgres/postgres/blob/REL_14_STABLE/src/backend/optimizer/plan/planner.c#L3057

**Review by yjhjstz — APPROVED** [review]
LGTM

**Comment by avamingli** [comment]
Pushed, thanks for review.

**Pull request merged**